designs/2019-additional-lint-targets/README.md

```diff
@@ -0,0 +1,133 @@
+- Start Date: 2019-05-12
+- RFC PR:
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring Additional Lint Targets with `.eslintrc`
+
+## Summary
+
+This proposal adds the ability to specify additional target files into configuration files. This enhancement will solve the pain that people have to use the `--ext` option with wanted file extensions even if they use plugins which support additional file types.
+
+## Motivation
+
+People have to use the `--ext` option or glob patterns to check wanted files even if they use plugins which support additional file types.
+
+```yml
+plugins:
+  - markdown
+  - html
+  - "@typescript-eslint"
+  - react
+  - vue
+```
+
+```bash
+# ESLint checks only `*.js`.
+eslint src docs
+
+# Needs `--ext` option
+eslint src docs --ext .js,.md,.html,.ts,.jsx,.tsx,.vue
+```
+
+However, using plugins which support additional file types is the intention that wants to check those files. The requirement of redundant CLI options is not reasonable.
+
+Per "[Related Discussions](#related-discussions)" section, this is very regularly requested; I want to configure additional target files with `.eslintrc`.
+
+## Detailed Design
+
+This proposal enhances `overrides` property of the config file.
+
+- If a config file in a directory has `overrides` property, ESLint checks the files which are matched by any of override entries (i.e., `files`/`excludedFiles` criteria) additionally in the directory.
+    - If any of `files` value of a override entry ended with `*`, ESLint ignores the override entry to avoid checking all files.
+    - This enhancement affects only the directory path. If people gave glob patterns, ESLint behaves the same as currently.
+    - The ignoring configuration (`.eslintignore`) is prior to this enhancement. If `.eslintignore` contains the additional target files, ESLint just ignores those as same as currently.
+
+The `overrides` property means that people intend to check those files. So this behavior is intuitive.
```

Comment:
What happens if I use `--ext=.js,.jsx` and I also have a `files` section for `.html`? Would they all be combined (so `.js`, `.jsx`, and `.html` files would be linted)? Or would `ext` take precedence (so `.html` would not be linted)?

Reply:
Oh, I didn't consider such a case.

I think that ESLint uses only `--ext` option if it was given. (`.html` would not be linted)